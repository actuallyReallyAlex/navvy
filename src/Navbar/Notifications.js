import React from 'react'
import { Box, Menu, Text } from 'grommet'
import { MailOption, Notification, Server } from 'grommet-icons'

const NavbarNotifications = () => {
  const notificationIsHovered = false; // ! Fix this

  return (
    <Box margin={{ horizontal: 'small' }}>
      <Menu
        dropAlign={{ top: 'bottom', right: 'right' }}
        icon={
          <Notification color={notificationIsHovered ? 'dark-2' : 'dark-6'} />
        }
        items={[
          {
            label: (
              <Box direction="row" gap="small" pad="small">
                <MailOption />
                <Text>You have 6 new messages</Text>
              </Box>
            ),
            onClick: () => {}
          },
          {
            label: (
              <Box direction="row" gap="small" pad="small">
                <Server />
                <Text>Server rebooted</Text>
              </Box>
            ),
            onClick: () => {}
          },
          {
            label: (
              <Box border="top" direction="row" pad="small">
                <Text className="spaced-text" size="small" weight="bold">
                  VIEW ALL NOTIFICATIONS
                </Text>
              </Box>
            )
          }
        ]}
      />
    </Box>
  )
}

export default NavbarNotifications
