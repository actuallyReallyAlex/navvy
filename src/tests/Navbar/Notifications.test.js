import React from 'react'
import Notifications from '../../Navbar/Notifications'
import { shallow } from 'enzyme'

const defaultProps = {}

describe('<Notifications />', () => {
  afterEach(() => {})

  test('Should render the <Notifications /> component.', () => {
    const component = shallow(<Notifications {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
