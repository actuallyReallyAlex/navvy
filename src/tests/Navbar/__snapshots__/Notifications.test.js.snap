// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notifications /> Should render the <Notifications /> component. 1`] = `ShallowWrapper {}`;
